tests/scenarios/single_entity_pipelines/test_single_entity_pipeline.py

```diff
@@ -0,0 +1,177 @@
+from wyvern.entities.model_entities import ModelOutput
+from wyvern.entities.request import BaseWyvernRequest
+from wyvern.service import WyvernService
+
+
+class Seller(WyvernEntity):
+    seller_id: str
+
+    def generate_identifier(self) -> Identifier:
+        return Identifier(
+            identifier=self.seller_id,
+            identifier_type="seller",
+        )
+
+
+class Buyer(WyvernEntity):
+    buyer_id: str
+
+    def generate_identifier(self) -> Identifier:
+        return Identifier(
+            identifier=self.buyer_id,
+            identifier_type="buyer",
+        )
+
+
+class Order(WyvernEntity):
+    order_id: str
+
+    def generate_identifier(self) -> Identifier:
+        return Identifier(
+            identifier=self.order_id,
+            identifier_type="order",
+        )
+
+
+class FraudRequest(BaseWyvernRequest):
+    seller: Seller
+    buyer: Buyer
+    order: Order
+
+
+class FraudResponse(SingleEntityPipelineResponse[float]):
+    reasons: List[str]
+
+
+class FraudRuleModel(SingleEntityModelComponent[FraudRequest, ModelOutput[float]]):
+    async def inference(self, input: FraudRequest, **kwargs) -> ModelOutput[float]:
+        return ModelOutput(
+            data={
+                input.order.identifier: 1,
+            },
+        )
+
+
+class FraudAssessmentModel(
+    SingleEntityModelComponent[FraudRequest, ModelOutput[float]],
+):
+    async def inference(self, input: FraudRequest, **kwargs) -> ModelOutput[float]:
+        return ModelOutput(
+            data={
+                input.order.identifier: 1,
+            },
+        )
+
+
+fraud_model = SingleEntityModelChain[FraudRequest, ModelOutput[float]](
+    FraudRuleModel(),
+    FraudAssessmentModel(),
+    name="fraud_model",
+)
+
+
+class FraudBusinessLogicComponent(
+    SingleEntityBusinessLogicComponent[float, FraudRequest],
+):
+    async def execute(
+        self,
+        input: SingleEntityBusinessLogicRequest[float, FraudRequest],
+        **kwargs,
+    ) -> float:
+        if input.request.seller.identifier.identifier == "test_seller_new":
+            return 0.0
+        return input.model_output
+
+
+fraud_biz_pipeline = SingleEntityBusinessLogicPipeline(
+    FraudBusinessLogicComponent(),
+    name="fraud_biz_pipeline",
+)
+
+
+class FraudPipeline(SingleEntityPipeline[FraudRequest, float]):
+    PATH = "/fraud"
+    REQUEST_SCHEMA_CLASS = FraudRequest
+    RESPONSE_SCHEMA_CLASS = FraudResponse
+
+    def generate_response(
+        self,
+        input: FraudRequest,
+        pipeline_output: float,
+    ) -> FraudResponse:
+        if pipeline_output == 0.0:
+            return FraudResponse(
+                data=pipeline_output,
+                reasons=["Fraudulent order detected!"],
+            )
+        return FraudResponse(
+            data=pipeline_output,
+            reasons=[],
+        )
+
+
+fraud_pipeline = FraudPipeline(model=fraud_model, business_logic=fraud_biz_pipeline)
```

Comment:
this is the "new experience" of passing the model and business_logic to the pipeline to define the pipeline.

Does it look simpler than the current RankingPipeline.get_model pattern?
```
def get_model(self) -> ModelCompolent:
    return SomeModelComponent()
```

Reply:
@ykeremy i think the `model=xxx` pattern definitely makes it easier to play around with different version of models in the future for a pipeline. i prefer `model=xxx` pattern.  wdyt?